mitodl/ocw-hugo-themes
# add ocw-course-v3 and course-offline-v2

#### Pre-Flight checklist


#### What are the relevant tickets?
Part of https://github.com/mitodl/ocw-hugo-themes/issues/1013

#### What's this PR do?
This PR adds the `course-v3` and `course-offline-v2` themes to the repo, which are duplicates of `course-v2` and `course-offline` respectively. This is being done to facilitate moving to rendering a course site using relative URLs and removing URL overrides in `course-offline` in a non-destructive manner.

#### How should this be manually tested?
 - Clone the `cg/course-relative` branch of `ocw-hugo-projects` (https://github.com/mitodl/ocw-hugo-projects/pull/244)
 - Point to the Hugo configs in the `ocw-course-v3` project in your env with `COURSE_HUGO_CONFIG_PATH=/path/to/ocw-hugo-projects/ocw-course-v3/config.yaml`
 - Run `yarn start course` to start a course of your choice
 - Inspect the various links in the nav, course title banner, etc. and make sure they are relative, clicking around the site to make sure they work
 - Switch your env back to `COURSE_HUGO_CONFIG_PATH=/path/to/ocw-hugo-projects/ocw-course-v2/config.yaml`
 - Restart the course site
 - If you inspect the links again, they should be back to being absolute URLs

#### Any background context you want to provide?
The following functionality is out of scope of this PR, and the reason we are creating a separate theme to get this work done is to be able to get the ball rolling without breaking these features in `course-v2`:

 - Links to other courses i.e. `/courses/course-1`
 - The sitemap

These features don't work on the local dev server anyway, but I figured it was worth noting here.
